sw/device/silicon_creator/lib/drivers/pinmux.c

```diff
@@ -121,9 +113,14 @@ void pinmux_init(void) {
       otp_read32(OTP_CTRL_PARAM_OWNER_SW_CFG_ROM_BOOTSTRAP_EN_OFFSET);
   if (launder32(bootstrap_en) == kHardenedBoolTrue) {
     HARDENED_CHECK_EQ(bootstrap_en, kHardenedBoolTrue);
-    configure_strap_pin(kInputSwStrap0);
-    configure_strap_pin(kInputSwStrap1);
-    configure_strap_pin(kInputSwStrap2);
+    // Note: attributes should be configured before the pinmux matrix to avoid
+    // "undesired electrical behavior and/or contention at the pads".
+    enable_pull_down(kInputSwStrap0.pad);
+    enable_pull_down(kInputSwStrap1.pad);
+    enable_pull_down(kInputSwStrap2.pad);
+    configure_input(kInputSwStrap0);
+    configure_input(kInputSwStrap1);
+    configure_input(kInputSwStrap2);
```

Comment:
nit: can't just reverse the order in which `configure_input()` and `enable_pull_down()` was called in `configure_strap_pin()`?

Reply:
The reason I did it this way is that I wasn't sure from the pinmux docs whether it meant "configure the attributes for each pad before configuring the pinmux matrix _for that pad_" or whether it meant "all attributes should be configured before anything in the pinmux matrix is configured" -- so I went with the most conservative interpretation.

Reply:
tagging @msfschaffner - can you look at the text in pinmux/doc/_index.md at/around line 289 and maybe clarify what is actually required?  (see @jadephilipoom comment above about uncertainty)

Reply:
Both approaches are fine. The important part is that the attributes for a pad are configured before the pinmux matrix for that pad is configured.